include/wx/generic/dirctrlg.h

```diff
@@ -115,6 +131,15 @@ class WXDLLIMPEXP_CORE wxGenericDirCtrl: public wxControl
     void OnTreeSelChange(wxTreeEvent &event);
     void OnItemActivated(wxTreeEvent &event);
     void OnSize(wxSizeEvent &event );
+    void OnRightClick(wxTreeEvent& event);
+
+    // Add a new menu item to the popup right-click menu
+    int NewMenuItem(const wxString& label);
+    wxMenu* GetMenu() { return m_rightClickMenu; }
+    wxDirItemData* GetRightClickItemData()
```

Comment:
I think we should remove this one too and just make `GetItemData()` public.

Reply:
And also provide access to 'm_rightClickedItemId' ?

Reply:
There would even be no need for it if the "menu popping up" event were `wxTreeEvent` and so carried the item it applied to... But OTOH there is no real harm in having some `GetItemShowingPopupMenu()` or something similar, so why not.

Reply:
I'm not sure which option you preferred:
1. Make `m_rightClickedItemId` public.
2. Leave `GetRightClickItemData()` in place.
3. Change the event to `wxTreeEvent`.

Reply:
I think the event should be changed to `wxTreeEvent` no matter what, just for consistency with the other events (and also because I think it would make sense to allow vetoing it, which is impossible with just a `wxCommandEvent`).

I think we might not need anything else. But if you think having `GetPopupMenuItem()` would be useful, we can add it. We definitely don't want to make any fields public.

Reply:
Hang on, there are two different events here. Let me be sure I'm doing the right thing.

Firstly there's the "Menu popped up" event, which can be a tree event, and so carry the tree item it applied to. This event is used by the user code to append items to the menu. 

Secondly, there's the event generated when the user selects an option from the pop-up menu. I assume that can't be a `wxTreeEvent`?  That then calls a function which needs the tree item, and that function would call `GetRightClickItemData()`.

So, I'm thinking I should make the first event into a `wxTreeEvent`, and leave `GetRightClickItemData()`, but rename it to `GetMenuItemData()`.

Reply:
In principle, you could stash the item you need in the `wxEVT_DIRCTRL_SHOWING_POPUP_MENU` handler and use it when processing `wxCommandEvent` from the menu item, but you're right, it's indeed simpler to just get the item from `wxDirCtrl` itself. So let's keep this function, but please let's have a `GetPopupMenuItem()` rather than `Data()`, it's unnatural to have access only to the data attached to the item, but not the item itself.